bin/reth/src/cli/mod.rs

```diff
@@ -84,6 +84,15 @@ impl<Ext: RethCliExt> Cli<Ext> {
 
         let _guard = self.init_tracing()?;
 
+        // Runtime safety check for the `optimism` feature flag being enabled on a non-OP Stack
+        // chain.
+        #[cfg(feature = "optimism")]
+        if !self.chain.is_optimism() {
+            eyre::bail!(
+                "The `optimism` feature flag is enabled, but the chain is a non OP Stack chain."
+            )
+        }
```

Comment:
this needs be moved to the reth main, so we can use --version or --help to verify this in CI.